assertk/src/commonMain/kotlin/assertk/assertions/charsequence.kt

```diff
@@ -53,3 +54,122 @@ fun Assert<CharSequence>.hasSameLengthAs(other: CharSequence) = given { actual -
     if (actualLength == otherLength) return
     expected("to have same length as:${show(other)} ($otherLength) but was:${show(actual)} ($actualLength)")
 }
+
+/**
+ * Asserts the char sequence has the expected number of lines.
+ */
+fun Assert<CharSequence>.hasLineCount(lineCount: Int) = given { actual ->
+    val actualLineCount = actual.lines().size
+    if (actualLineCount == lineCount) return
+    expected("to have line count:${show(lineCount)} but was:${show(actualLineCount)}")
+}
+
+/**
+ * Asserts the char sequence is equal to the expected char sequence.
+ * @param ignoreCase true to compare ignoring case, the default if false.
+ * @see [isNotEqualTo]
+ */
+fun Assert<CharSequence?>.isEqualTo(other: CharSequence?, ignoreCase: Boolean = false) = given { actual ->
+    val actualString = actual?.toString()
+    val otherString = other?.toString()
+    if (actualString.equals(otherString, ignoreCase)) return
```

Comment:
So it's possible for a `CharSequence` to not be equal but their `toString()` representations to be equal (Think spanned strings on android). Think we need to keep these only for `String`.

Reply:
Good point :+1: